src/purple_vm/src/virtual_machine.rs

```diff
@@ -3472,6 +3507,180 @@ fn perform_max(op: Instruction, operand_stack: &mut Stack<VmValue>) {
     };
 }
 
+fn perform_float_common(op: Instruction, operand_stack: &mut Stack<VmValue>) -> Result<(), VmError> {
+    let len = operand_stack.len();
+    
+    fn is_type_float(operand: VmValue) -> bool {
+        match operand{
+            VmValue::I32(_) | VmValue::I64(_) | 
+            VmValue::i32Array2(_) | VmValue::i32Array4(_) | VmValue::i32Array8(_) | 
+            VmValue::i64Array2(_) | VmValue::i64Array4(_) | VmValue::i64Array8(_) => false,
+            VmValue::F32(_) | VmValue::F64(_) | 
+            VmValue::f32Array2(_) | VmValue::f32Array4(_) | VmValue::f32Array8(_) | 
```

Comment:
This should match arrays up to 256 items in length, it seems that it only matches up to 8 now?